### PR TITLE
Adds an `InView` Predicate

### PR DESCRIPTION
**Pull request by nkumar-bdai**
This is needed in addition to `Reachable` for our planning to actually be correct.

### CODE REVIEW

File: predicators/envs/spot_env.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ class _SpotObservation:
     images: Dict[str, Image]
     # Objects that are seen in the current image and their positions in world
     objects_in_view: Dict[Object, Tuple[float, float, float]]
+    # Objects seen only by the hand camera
+    objects_in_hand_view: List[Object]
```
Comment (tsilver-bdai) [inline predicators/envs/spot_env.py:44]: can we make this a set?

File: predicators/perception/spot_bike_perceiver.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ class SpotBikePerceiver(BasePerceiver):
     def __init__(self) -> None:
         super().__init__()
         self._known_object_poses: Dict[Object, Tuple[float, float, float]] = {}
+        self._known_objects_in_hand_view: List[Object] = []
```
Comment (tsilver-bdai) [inline predicators/perception/spot_bike_perceiver.py:23]: ditto

**Review by tsilver-bdai — COMMENTED** [review]
one small thing and one medium thing communicated in person

**Comment by nkumar-bdai** [comment]
We decided to split this into multiple PR's (one just for InView, one for miscellaneous robot fixes, one for debugging the pick from floor thing, etc.)

**Review by tsilver-bdai — APPROVED** [review]

**Pull request merged**